helix-term/src/ui/picker.rs

```diff
@@ -30,11 +30,24 @@ type FileLocation = (PathBuf, Option<(usize, usize)>);
 pub struct FilePicker<T> {
     picker: Picker<T>,
     /// Caches paths to documents
-    preview_cache: HashMap<PathBuf, Document>,
+    preview_cache: HashMap<PathBuf, CaluculatedPreview>,
+    read_buffer: Vec<u8>,
     /// Given an item in the picker, return the file path and line number to display.
     file_fn: Box<dyn Fn(&Editor, &T) -> Option<FileLocation>>,
 }
 
+pub enum Preview<'c, 'e> {
+    OwenedPreview(&'c CaluculatedPreview),
+    EditorDocument(&'e Document),
+}
+/// Caluculated file preview
+pub enum CaluculatedPreview {
+    OwenedDocument(Document),
+    Binary,
+    LargeFile,
+    NotFound,
+}
```

Comment:
I have attached some other code down below which uses these variable names, so no particular reason to use these over what we currently have (brevity maybe ?)

```suggestion
pub enum Preview<'picker, 'editor> {
    Cached(&'picker CachedPreview),
    EditorDocument(&'editor Document),
}

pub enum CachedPreview {
    Document(Document),
    Binary,
    LargeFile,
    NotFound,
}
```

Reply:
I like this naming.